std/stdio.d

```diff
@@ -3793,13 +3794,82 @@ unittest
     }
 }
 
+// roll our own appender, but with "safe" arrays
+private struct ReadlnAppender
+{
+    char[] buf;
+    size_t cap;
+    size_t pos;
+    size_t initcap;
+
+    void initialize(char[] b)
+    {
+        buf = b;
+        cap = initcap = buf.capacity;
+        if (cap < buf.length)
+            cap = buf.length;
+        pos = 0;
+    }
+    @property char[] data()
+    {
+        // always shrink/extend the buffer to allow reuse in the next call
+        if (initcap > 0 || cap > buf.length)
```

Comment:
Seems like you could replace `cap` and `initcap` with a boolean flag `reuseBuf`.